week6/ro-el/src/components/InputBar.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { ChangeEvent, useState } from "react";
+import { Product } from "../App";
+import uuid from "react-uuid";
+
+interface Props {
+  product: Product;
+  addProduct: (product: Product) => void;
+  setIsEditBtnClicked?: React.Dispatch<React.SetStateAction<boolean>>;
+  editProduct?: (removeProduct: Product, newProduct: Product) => void;
+}
+
+const InputBar: React.FC<Props> = ({
+  product,
+  addProduct,
+  setIsEditBtnClicked,
+  editProduct,
+}) => {
+  const [newProduct, setNewProduct] = useState<Product>({
+    ...product,
+    id: uuid(),
+  });
+
+  const handleChange = (value: string | boolean, label: keyof Product) => {
+    setNewProduct({ ...newProduct, [label]: value });
+  };
+
+  const handleClickProductBtn = () => {
+    editProduct !== undefined
+      ? product.id !== newProduct.id &&
+        editProduct(product, { ...newProduct, id: uuid() })
+      : addProduct({ ...newProduct, id: uuid() });
```

Comment:
```suggestion
    editProduct 
      ? product.id !== newProduct.id &&
        editProduct(product, { ...newProduct, id: uuid() })
      : addProduct({ ...newProduct, id: uuid() });
```
editProduct ? 로 해도 동일합니다!

Reply:
boolean 타입이 아닌 값들은 
editProduct !== undefined로 하는게 더 안전하다고 합니다. 잘못알려드려 죄송합니다 ㅠ
edit Product !== null 로 하면 undefined와 null 둘다 걸러준다고 합니다.